ui/src/app/components/breadcrumb.tsx

```diff
@@ -190,13 +187,46 @@ export const Breadcrumb = fp.flow(
     }
 
     trail(): Array<BreadcrumbData> {
+      const workspaceMatch = matchPath<MatchParams>(location.pathname, {
```

Comment:
this is the most egregious use of `matchPath`, I think. Everything else is just one path. That said... this component is a big switch clause anyways

Reply:
this part does seem a bit too egregious. Would there be a way to query our location through the existing `AppRoute` and get the params out of that?

Or maybe we just need to think of a different approach altogether. @calbach might have some ideas?

The general case we're running up against is what to do if a component not within a `<Route>` needs to do something with route parameters.